applications/vanilla/settings/class.hooks.php

```diff
@@ -10,13 +10,23 @@
 
 use Garden\Container\Container;
 use Garden\Container\Reference;
-use Vanilla\Formatting\Embeds\EmbedManager;
 
 /**
  * Vanilla's event handlers.
  */
 class VanillaHooks implements Gdn_IPlugin {
 
+    /**
+     * Handle the container init event to register things with the container.
+     *
+     * @param Container $dic
+     */
+    public function container_init(Container $dic) {
```

Comment:
I don't think this is the best place to addProvider.

Why not in bootstrap.php?

Reply:
Because the class is from an addon (This is addon declaring it's contribution to the breadcrumb manager). If the forum application was disabled the `ForumBreadcrumbProvider` class would not be available. That's why the the registration lives in the addon initialization.